sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,343 @@
+    their length, and their material.
+
+    Parameters
+    ==========
+    length : Sympifyable
+        A SymPy expression representing the Beam's length.
+    elastic_modulus : Sympifyable
+        A SymPy expression representing the Beam's Modulus of Elasticity.
+        It is a measure of the stiffness of the Beam material.
+    second_moment : Sympifyable
+        A SymPy expression representing the Beam's Second moment of area.
+        It is a geometrical property of an area which reflects how its
+        points are distributed with regard to an arbitrary axis.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> Beam(1, E, I)
+    Beam(1, E, I)
+
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment):
+        self._length = length
+        self._elastic_modulus = elastic_modulus
+        self._second_moment = second_moment
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def length(self):
+        """Length of the Beam."""
+        return self._length
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    def apply_boundary_conditions(self, **bcs):
+        """
+        Takes the boundary conditions of the beam bending problem as input.
+        The boundary conditions should be passed as keyworded arguments.
+        It is suggested to use ``moment``, ``slope`` and ``deflection`` as the
+        keywords while providing the boundary conditions of the moment curve,
+        slope curve and the deflection curve respectively as inputs.
+
+        Outputs a dictionary.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> bcs = b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> bcs
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> bcs['moment']
+        [(0, 4), (4, 0)]
+
+        """
```

Comment:
This docstring doesn't yet explain what `[(0, 4), (4, 0)]` is. This is a list of two tuples and the numbers must represent something. What do they represent?